boshio/http_client.go

```diff
@@ -8,10 +8,30 @@ import (
 	"time"
 )
 
+func NewHTTPClient(host string, wait time.Duration) HTTPClient {
+	return HTTPClient{
+		Host: host,
+		Wait: wait,
+		Client: &http.Client{
+			Transport: &http.Transport{
+				Proxy: http.ProxyFromEnvironment,
+
+				Dial: (&net.Dialer{
+					Timeout:   30 * time.Second,
+					KeepAlive: 0, // don't send keepalive TCP messages
+				}).Dial,
+
+				TLSHandshakeTimeout: 60 * time.Second,
+				DisableKeepAlives:   true,
+			},
+		},
+	}
+}
+
 type HTTPClient struct {
 	Host   string
 	Wait   time.Duration
-	Client *http.Client
+	Client httpClient
```

Comment:
we used the interface here so that we could pass a fake in [this test](https://github.com/ljfranklin/bosh-io-stemcell-resource/blob/6658aaa9514c540094bc0819df1ecafd5e22cfaa/boshio/http_client_test.go#L90)